app/models/adviser.rb

```diff
@@ -32,6 +32,15 @@ class Adviser < ActiveRecord::Base
   validate :match_reference_number
 
   after_commit :geocode
+  after_save :check_for_changes
+
+  def self.move_to(receiving_firm)
+    self.transaction do
```

Comment:
No, this is what makes it work on a result set rather than just a single record. For a single record you just do `adviser.firm = new_firm`.